### PR TITLE
Fix MM-46309

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Fix MM-46309

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46309

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/utils/file/index.ts [hunk 1]
```diff
@@ -362,7 +362,7 @@ export function getLocalFilePathFromFile(serverUrl: string, file: FileInfo | Fil
                 }
             }
 
-            return `${FileSystem.CachesDirectoryPath}/${server}/${filename}-${hashCode(file.id!)}.${extension}`;
+            return `${FileSystem.CachesDirectoryPath}/${server}/${filename}-${file.id!}.${extension}`;
```
Comment (enahum) [inline app/utils/file/index.ts:365]: I think we should apply here the fix for https://mattermost.atlassian.net/browse/MM-46320 unless I'm mistaken and is not this one?

File: app/utils/security.ts [hunk 2]
```diff
@@ -7,19 +7,3 @@ export async function getCSRFFromCookie(url: string) {
     const cookies = await CookieManager.get(url, false);
     return cookies.MMCSRF?.value;
 }
-
-export const hashCode = (str: string): string => {
```
Comment (enahum) [inline app/utils/security.ts:11]: I wonder if we should keep this and add a deprecated comment.

The reason to keep it as cause we can use it to delete the previous database files and cache folders.

**Review by enahum — APPROVED** [review]
Overall looks good to me, left a couple of comments though to see if we take action

**Review by jupenur — CHANGES_REQUESTED** [review]
Just one thing: we need to use the URL-compatible variant of Base64 that doesn't contain slashes: https://datatracker.ietf.org/doc/html/rfc4648#section-5

It's a pretty trivial change though. Instead of `base64.encode(serverUrl)` we just need to do `base64.encode(serverUrl).replace(/\+/g, '-').replace(/\//g, '_')` everywhere.

**Comment by enahum** [comment]
Cool, perhaps a utility function to minimize the changes everywhere

File: app/utils/security.ts [hunk 3]
```diff
@@ -23,3 +25,7 @@ export const hashCode = (str: string): string => {
     }
     return hash.toString();
 };
+
+export const hashServerUrl = (str: string): string => {
```
Comment (jupenur) [inline app/utils/security.ts:29]: Nit: this isn't a hash. No reason to tie this to server URLs specifically either; there's a good chance we find more use cases for this type of encoding in the future.

A more accurate name would be something like `urlSafeBase64Encode`.

**Review by jupenur — APPROVED** [review]
Sorry for the delay. LGTM!

**Review by enahum — APPROVED** [review]

**Pull request merged**